src/pysignalr/__init__.py

```diff
@@ -44,10 +43,7 @@ async def __aiter__(
                 yield protocol
 
         # Handle expired connection URLs by raising a NegotiationTimeout exception.
-        except InvalidStatusCode as e:
-            if e.status_code == HTTPStatus.NOT_FOUND:
-                raise NegotiationTimeout from e
-        except asyncio.TimeoutError as e:
+        except (InvalidStatusCode,  asyncio.TimeoutError) as e:
             raise NegotiationTimeout from e
```

Comment:
My thinking was that all of them will require the same action on higher levels, i.e. retrieve the URL again and retry.
And there can be lots of different reasons for the server to respond with an error code, like 429 Too Many Requests for instance, which is what we have been seeing when the lib started trying to reconnect as fast as it could and made the server side say "stop, that is too much"...

But can I then suggest we call the 404 error NegotiationNotfound and all others NegotiationFailure? I mean, 404 is not a timeout, it is the server saying what you are asking for is not here (anymore). asyncio.TimeoutError can still be mapped to NegotiationTimeout.

Reply:
Yeah, you're right about 404 Not Found, it's not necessarily a 5s negotiation timeout:

https://learn.microsoft.com/en-us/azure/azure-signalr/signalr-howto-troubleshoot-guide

I guess, we can use `NegotiationFailure` for all causes and call it a day.

Reply:
Ok, good, I will make changes this evening.

Reply:
Ok, changes made, please look again.